File: python/해시/전화번호 목록.py

""" 레벨 2. 전화번호 목록 """

def solution(phone_book):
    
    phone_book.sort()

    for idx in range(len(phone_book)-1):
        prefix = phone_book[idx]
        if prefix == phone_book[idx+1][:len(prefix)]:
            return False

    return True

print(solution(["119", "97674223", "1195524421"]))      # false
print(solution(["123","456","789"]))                    # true
print(solution(["12","123","1235","567","88"]))         # false

File: Java/src/자료구조/전화번호목록.java
package 자료구조;

import java.util.Arrays;

public class 전화번호목록 {

    public static void main(String[] args) {

        String[] p1 = {"119", "97674223", "1195524421"};
        System.out.println(solution(p1));

        String[] p2 = {"123","456","789"};
        System.out.println(solution(p2));

        String[] p3 = {"12","123","1235","567","88"};
        System.out.println(solution(p3));
    }
    
    public static boolean solution(String[] phone_book) {
        
        Arrays.sort(phone_book);

        for (int i = 1; i < phone_book.length; i++) {
            if (phone_book[i].startsWith(phone_book[i-1])) {
                return false;
            }
        }
        
        return true;
    }
}
